### PR TITLE
Invalidate metadata cache when schema change happens in SQL API

### PR DESCRIPTION
**Pull request by brfrn169**
This PR adds a logic to invalidate metadata cache when schema change happens in SQL API. I will inline comments to make it more readable. Please take a look!

### CODE REVIEW

File: core/src/main/java/com/scalar/db/sql/DdlStatementExecutor.java [hunk 1]
```diff
@@ -23,9 +26,11 @@
 public class DdlStatementExecutor implements DdlStatementVisitor<Void, Void> {
```
Comment (brfrn169) [inline core/src/main/java/com/scalar/db/sql/DdlStatementExecutor.java:26]: The cache invalidations are kicked when a schema change happens in `DdlStatementExecutor`.

File: core/src/main/java/com/scalar/db/sql/metadata/CachedMetadata.java [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+import com.scalar.db.api.DistributedTransactionAdmin;
+import com.scalar.db.exception.storage.ExecutionException;
+import com.scalar.db.sql.exception.SqlException;
+import java.util.Optional;
+import java.util.concurrent.TimeUnit;
+import javax.annotation.Nonnull;
+import javax.annotation.concurrent.ThreadSafe;
+
+@ThreadSafe
+public class CachedMetadata implements Metadata {
```
Comment (brfrn169) [inline core/src/main/java/com/scalar/db/sql/metadata/CachedMetadata.java:15]: I also extract metadata related interfaces in this PR.

The extracted interfaces are `Metadata`, `NamespaceMetadata`, `TableMetadata`, `ColumnMetadata`, and `IndexMetadata`.

File: core/src/main/java/com/scalar/db/sql/DdlStatementExecutor.java [hunk 3]
```diff
@@ -125,6 +142,17 @@ public Void visit(DropNamespaceStatement statement, Void context) {
   public Void visit(DropTableStatement statement, Void context) {
     try {
       admin.dropTable(statement.namespaceName, statement.tableName, statement.ifExists);
+
+      // Invalidate the metadata cache
+      metadata
```
Comment (feeblefakie) [inline core/src/main/java/com/scalar/db/sql/DdlStatementExecutor.java:147]: Should it be done in the finally clause to care for the case where dropTable fails for some reason?
(it's a minor case, but if `dropTable` fails after dropping the specified table, the corresponding metadata will be left un-invalidated?)

**Review by feeblefakie — COMMENTED** [review]
Looking good!
I left one question.

File: core/src/main/java/com/scalar/db/sql/DdlStatementExecutor.java (hunk 3 above)
Comment (brfrn169) [reply core/src/main/java/com/scalar/db/sql/DdlStatementExecutor.java:147]: Thank you for taking a look at this! Yes, that's better. I will modify it in that way. Thanks.

**Review by Torch3333 — APPROVED** [review]
LGTM, thank you.

**Review by feeblefakie — APPROVED** [review]
LGTM! Thank you!

**Pull request merged**